kerneldriver_linux.go

```diff
@@ -0,0 +1,68 @@
+// +build linux
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// File originally made by jpoirier (https://github.com/jpoirier)
+// Addapted to the google/gousb library by nicovell3
+
+package gousb
+
+import (
+	"fmt"
+)
+
+/*
+#cgo pkg-config: libusb-1.0
+#include <libusb.h>
+*/
+import "C"
+
+// detachKernelDriver detaches any active kernel drivers, if supported by the platform.
+// If there are any errors, like Context.ListDevices, only the final one will be returned.
+func (d *Device) detachKernelDriver() (err error) {
+	for _, cfg := range d.Desc.Configs {
+		for _, iface := range cfg.Interfaces {
+			switch activeErr := C.libusb_kernel_driver_active((*C.libusb_device_handle)(d.handle), C.int(iface.Number)); activeErr {
```

Comment:
I don't think you need a two stage process. You could instead just call detach_kernel_driver blindly, if it returns NOT_FOUND, then there was no driver attached.